pkg/cmd/cli/cmd/newapp.go

```diff
@@ -483,6 +483,9 @@ func setupAppConfig(f *clientcmd.Factory, out io.Writer, c *cobra.Command, args
 	if len(config.SourceImage) == 0 && len(config.SourceImagePath) != 0 {
 		return kcmdutil.UsageError(c, "--source-image must be specified when --source-image-path is specified.")
 	}
+	if len(config.Strategy) != 0 && config.Strategy != "docker" && config.Strategy != "source" {
```

Comment:
I would rather make "docker" and "source" constants in pkg/build/api/types.go (or make a TODO for that ;-)